src/main/java/org/datadog/jenkins/plugins/datadog/model/BuildData.java

```diff
@@ -261,8 +265,25 @@ private void populateEnvVariables(EnvVars envVars){
      * @param envVars
      */
     private void populateGitVariables(Run<?,?> run, TaskListener listener, EnvVars envVars) {
-        final GitCommitAction gitCommitAction = GitUtils.buildGitCommitAction(run, listener, envVars, this.gitCommit, this.nodeName, this.workspace);
-        if(gitCommitAction != null) {
+        // First we obtain the actions to check if the Git information was already calculated.
+        // If so, we want to use this information to avoid creating a new Git client instance
+        // to calculate the same information.
+        GitCommitAction gitCommitAction = run.getAction(GitCommitAction.class);
+        GitRepositoryAction gitRepositoryAction = run.getAction(GitRepositoryAction.class);
```

Comment:
I understand the strategy here, but I don't understand where's the code that calls `.addAction` with the git client. Can you point me to that place?

Reply:
Sure. The methods are `GitUtils.buildGitRepositoryAction(...)` and `GitUtils.buildGitCommitAction(...)`